File: python/proteomaps_PATHNAMES.py
import os

class proteomaps_PATHNAMES:

  def __init__(self, hierarchy_version):

    # Base and tmp directory -- please edit
    self.BASE_DIR              = "/home/wolfram/Proteomaps/"
    self.TMP_DIR               = "/home/wolfram/tmp/matlab_protein_abundance";

    # Predefined directory names
    self.PROTEIN_HIERARCHY_DIR = self.BASE_DIR + "genomic_data/" + hierarchy_version + "/"
    self.PROTEIN_LENGTH_DIR    = self.PROTEIN_HIERARCHY_DIR + "protein_length_data/"

    # Files / subdirectories within the functional hierarchy
    self.INFILE_KO_HIERARCHY_FILE  = self.PROTEIN_HIERARCHY_DIR + "/KO_gene_hierarchy.tms"
    self.INFILE_ANNOTATION_CHANGES = self.PROTEIN_HIERARCHY_DIR + "/KO_gene_hierarchy_changes.csv"
    self.INDIR_ORGANISM_MAPPING    = self.PROTEIN_HIERARCHY_DIR + "/KO_gene_hierarchy_organism_mapping/"
    self.ORGANISM_INFORMATION      = self.PROTEIN_HIERARCHY_DIR + "/organisms.csv"

    # Python directory
    self.WORKFLOW_PATH = self.BASE_DIR + 'python/proteomaps_workflow/'
    self.MATLAB_PATH   = self.BASE_DIR + 'matlab/'

File: matlab/proteomaps_path_names.m
function pathnames = proteomaps_path_names(data_directory)

pathnames.BASE_DIR              = [fileparts(which(mfilename)) filesep '..' filesep];
pathnames.PROTEIN_LENGTH_DIR    = [pathnames.BASE_DIR '/genomic_data/protein_length_data/'];
pathnames.PROTEOMAPS_MATLAB_DIR = [pathnames.BASE_DIR '/matlab'];

pathnames.TMP_DIR  = [tempdir '/matlab_protein_abundance'];
[~,~] = mkdir(pathnames.TMP_DIR);

if ~exist('hierarchy_version','var'),
  pathnames.hierarchy_version = 'KO_gene_hierarchy_2015-01-01';
end

if ~exist('RESOURCE_DIR','var'),
  pathnames.RESOURCE_DIR       = [pathnames.BASE_DIR 'genomic_data/' pathnames.hierarchy_version '/'];
end

pathnames.PROTEOMAPS_RESOURCE_DIR = pathnames.RESOURCE_DIR;
pathnames.SUBSAMPLED_TREE_DIR     = pathnames.TMP_DIR;

pathnames.PROTEOMAPS_MATLAB_DATA_DIR = [data_directory '/tables/'];
